public/modules/custom/helfi_audit_log/src/ResilientLoggerTasks.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\helfi_audit_log;
+
+use Psr\Log\LoggerInterface;
+use ResilientLogger\ResilientLogger;
+use Drupal\Core\State\StateInterface;
+
+class ResilientLoggerTasks {
+  private const SETTINGS_NAME = "resilient_logger";
+  private const LOGGER_CHANNEL = "resilient_logger.tasks";
+  private const SERVICE_NAME = "resilient_logger.service";
+
+  private const DEFAULT_OFFSET_SUBMIT = "+15min";
+  private const DEFAULT_OFFSET_CLEAR = "first day of next month midnight";
+
+  private const PARAM_KEY_OFFSET_SUBMIT = "schedule_submit_unsent_entries";
+  private const PARAM_KEY_OFFSET_CLEAR = "schedule_clear_sent_entries";
+
+  private const STATE_KEY_PREV_SUBMIT = "resilient_logger.prev_submit_unsent";
+  private const STATE_KEY_PREV_CLEAR = "resilient_logger.prev_clear_sent";
+
+  private string $submitDateOffset;
+  private string $clearDateOffset;
+  private StateInterface $state;
+  private LoggerInterface $logger;
+  private ResilientLogger $service;
+
+  /**
+   * Drupal-specific helper for scheduling ResilientLogger tasks.
+   * Tasks are not executed immediately but are evaluated during Drupal's cron runs.
+   * Each task runs conditionally based on time offsets and the last execution time,
+   * not on every cron trigger.
+   * @see https://www.php.net/manual/en/function.strtotime.php
+   * 
+   * @param string $submitDateOffset
+   *   String representation of the next time submit unsent entries task should run.
+   *   Defaults to +15min from previous one.
+   * @param string $clearDateOffset
+   *   String representation of the next time clear old entries task should run.
+   *   Defaults to first day of the next month at 00:00.
+   * 
+   */
+  public function __construct(
+    ?string $submitDateOffset,
+    ?string $clearDateOffset
```

Comment:
Use constructor property promotion for properties: https://stitcher.io/blog/constructor-promotion-in-php-8.